### ISSUE TITLE
Segfault in GCM for x86_64 without hardware acceleration

### ISSUE THREAD

**Issue by jvdsn** [openssl/openssl#19673]
Context: https://github.com/openssl/openssl/blob/master/crypto/modes/gcm128.c#L414-L424

When `GHASH_ASM` is defined, `ctx->ghash` is set to NULL. However, when hardware acceleration is not available on x86_64 (e.g. without PCLMULQDQ), `ctx->ghash` stays NULL. This causes a segfault when trying to use GCM. This can be emulated by using the `OPENSSL_ia32cap` variables as follows:
```
$ openssl version
OpenSSL 3.2.0-dev  (Library: OpenSSL 3.2.0-dev )
$ OPENSSL_ia32cap=~0x1240000200000000:~0x00000600C0230000 openssl speed -evp aes-128-gcm
Doing AES-128-GCM for 3s on 16 size blocks: zsh: segmentation fault (core dumped)  OPENSSL_ia32cap=~0x1240000200000000:~0x00000600C0230000 openssl speed -evp 
```

This seems related to https://github.com/openssl/openssl/commit/186be8ed26f5561faf91d6da3ed14cd9cb6617dd where it was fixed for ARM and SPARC?

**Comment by t8m** [comment]
Does not apply to 3.0. Will apply to 3.1 after the gcm refactoring is applied there.

**Comment by t8m** [comment]
Fix in #19674
